### ISSUE TITLE
Shadow Augmentation 

### ISSUE THREAD

**Issue by bluesky314** [PeterL1n/BackgroundMattingV2#59]
The paper mentions that shadows were artificially added to make the model more robust to such situations. Using what method were these shadows created and will the code be shared for it?

**Comment by PeterL1n** [comment]
Take the ground-truth alpha, randomly blur it, and perform some affine transformation. Then you can darken the background with this alpha mask as it was a shadow cast by the subject.

Note this darkened background is used to composite with the foreground to become the source image, the original background is provided to the network as the background input. This simulates that the pre-captured background doesn't have subject nor shadow, and the source image has subject and shadow.

**Comment by PeterL1n** [comment]
The training code will be released when the paper is accepted.
